lib/beacon/router.ex

```diff
@@ -185,6 +185,9 @@ defmodule Beacon.Router do
   Then Beacon will reference both of those sitemaps in the top-level index:
     * `my_domain.com/sitemap_index.xml`
 
+  Note that `beacon_sitemap_index` will include the sitemap URL of all mounted sites
+  in the router, so that macro should be at the root and not duplicated.
+
```

Comment:
what does it mean that it should be at the root?  is it still OK for e.g.
```elixir
scope "/admin", as: :beacon_live_admin do
  pipe_through [:admin_browser, ...]

  get "/session/sign_out", SessionController, :sign_out

  beacon_live_admin "/"
end

scope "/" do
  pipe_through [:browser, ...]
  beacon_site "/", site: :dockyard_com
end
```

Reply:
ok just checking there wasn't an additional meaning that needed documentation

Reply:
Wait... giving a second thought we can't render all sites in the sitemap index. Giving 2 different scopes on different `:host` it should include only the sites within that scope.

Reply:
that would mean multiple sitemap indexes, right?  we don't currently have a way to do that 🤔

Reply:
Eg: 

```
scope host: foo.com
  beacon_site :site_a  
  beacon_site :site_b 
  sitemap_index   

scope host: bar.com
  beacon_site :site_c 
  sitemap_index
```

Should foo.com/sitemap_index.xml include site_c?

Reply:
I think in order to properly implement this, we need some data model "in-between" 
```
[otp_app] --- [ ??? ] --- [Beacon :site]
```
where that `???` would represent a "host"

Reply:
`sitemap.xml` and `robots.txt` scoped by site are fine, served at `{host}/{prefix}/{file}`, but the sitemap index it seems like should only include the sites running in the same scope where it's defined or have an option to define which sites should be included in the index.

So either an option like:

```
scope host: foo.com
  beacon_site :site_a  
  beacon_site :site_b 
  sitemap_index [:site_a, :site_b]

scope host: bar.com
  beacon_site :site_c 
  sitemap_index [:site_c]
```

_Where the default could be "all" sites._

Or introspect the router to accumulate sites per scope, similar to how we accumulate sites in `__beacon_sites__` in the router.

Although that's not responsibility of this PR.